# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/lib/wasm.js
# Commit: 13b7086f65cb
# Full Hash: 13b7086f65cbbbf1fed6e0341cfbdfb309fb962c
# Author: Julien Pages <jpages@mozilla.com>
# Date: 2025-09-17 13:36 +0000
# Regressor Bug: 1969509
# File Overlap Count: 6
# Overlapping Files: js/src/vm/ArrayBufferObject.cpp, js/src/wasm/WasmMemory.h, js/src/jit-test/tests/wasm/memory64/basic.js, js/src/jit-test/tests/wasm/wasm-resizablearraybuffer.js, js/src/jit-test/lib/wasm.js
# Description:
#   Bug 1969509 - wasm: Update .maxByteLength to report the source wasm max pages in bytes. r=rhunt
# ==============================================================================

diff -r f7c96ac82c1a -r 13b7086f65cb js/src/jit-test/lib/wasm.js
--- a/js/src/jit-test/lib/wasm.js	Wed Sep 17 13:35:00 2025 +0000
+++ b/js/src/jit-test/lib/wasm.js	Wed Sep 17 13:36:10 2025 +0000
@@ -519,6 +519,12 @@
 // Max number of memories in a single wasm module.
 let MaxMemories = 100
 
+// Constants related to memory sizes.
+const MaxMemory64PagesValidation = BigInt(Math.pow(2, 37) - 1); // from spec
+const MaxTable64ElemsValidation = 0xFFFF_FFFF_FFFF_FFFFn; // from spec
+const MaxTableElemsRuntime = 10000000; // from WasmConstants.h
+const MaxUint32 = 0xFFFF_FFFF;
+
 // Common array utilities
 
 // iota(n,k) creates an Array of length n with values k..k+n-1